packages/giselle/src/react/act/use-act-controller.tsx

```diff
@@ -16,18 +16,25 @@ type CreateAndStartActParams = Omit<
 		| "applyPatches"
 		| "generationAdapter"
 		| "onActStart"
-		| "applyPatches"
+		| "onActComplete"
 		| "startGeneration"
 	> & {
-		onActStart: (cancel: () => Promise<void>) => void | Promise<void>;
+		onActStart?: (
+			cancel: () => Promise<void>,
+			actId: string,
+		) => void | Promise<void>;
+		onActComplete?: (
+			hasError: boolean,
+			duration: number,
+			actId: string,
+		) => void | Promise<void>;
```

Comment:
This is not very clear, so I would like to make it an object.

```suggestion
		onActStart?: (options: {
			cancel: () => Promise<void>,
			actId: string,
		}) => void | Promise<void>;
		onActComplete?: (options: {
			hasError: boolean,
			duration: number,
			actId: string,
		}) => void | Promise<void>;
```